packages/libvgm/pthread_cancel.patch

```diff
@@ -0,0 +1,16 @@
+diff -uNr libvgm/utils/OSThread_POSIX.c libvgm.mod/utils/OSThread_POSIX.c
+--- libvgm/utils/OSThread_POSIX.c	2022-12-16 17:19:46.508761717 +0800
++++ libvgm.mod/utils/OSThread_POSIX.c	2022-12-16 17:26:29.958761563 +0800
+@@ -78,8 +78,11 @@
+ 	
+ 	if (! thr->id)
+ 		return;
+-	
++#ifndef __ANDROID__	
+ 	retVal = pthread_cancel(thr->id);
++#else
++	retVal = pthread_kill(thr->id, SIGUSR2);
++#endif
```

Comment:
This is not correct.

Reply:
More precisely, this does not work as expected unless the thread changes the action for `SIGUSR2` by using `sigaction(2)`.

Reply:
So should I use `sig`?

Reply:
I would say you are not qualified to package this, unfortunately.

Reply:
To elaborate I guess @xtkoba means that a signal-catching function needs to be set for SIGUSR2, as done for example in [@xtkoba's calcurse patch](https://github.com/termux/termux-packages/blob/master/packages/calcurse/threading.patch#L7-L29)